addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRuleUnitTest.java

```diff
@@ -249,6 +317,80 @@ protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
         assertThat(alertsRaised.get(0).getEvidence().startsWith(HttpHeader.HTTP), equalTo(true));
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {HttpFieldsNames.LOCATION, HttpFieldsNames.REFRESH})
+    void shouldNotReportRedirectWithMetaLocationOrRefreshWhenConcatenatedInParam(String type)
+            throws Exception {
+        // Given
+        String test = "/shouldNotReportRedirectWithMetaLocationOrRefreshWhenConcatenatedInPath/";
+
+        nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        String site = getFirstParamValue(session, "site");
+                        if (site != null && site.length() > 0) {
+                            Response response =
+                                    newFixedLengthResponse(
+                                            NanoHTTPD.Response.Status.OK,
+                                            NanoHTTPD.MIME_HTML,
+                                            "<html><head><meta http-equiv=\""
+                                                    + type
+                                                    + "\" content=\""
+                                                    + "https://example.com/?q=\""
```

Comment:
Was the double quote after the query parameter added on purpose? That makes the HTML malformed.

Reply:
No that must be a mistake, I must have lost track of pairs. I believe that one is a duplicate closing of the `content` attribute. Will fix.